miasm2/arch/x86/sem.py

```diff
@@ -1777,9 +1777,11 @@ def movs(ir, instr, size):
 def movsd(_, instr, dst, src):
     e = []
     if isinstance(dst, m2_expr.ExprId) and isinstance(src, m2_expr.ExprMem):
-        src = m2_expr.ExprMem(src.arg, dst.size)
+        src = m2_expr.ExprMem(src.arg, 64)
+        dst = dst[:64]
```

Comment:
You're completely right, I've just tested on a real CPU

I'll update the PR accordingly